data/src/test/kotlin/com/antosik/benchproject/data/popular/movies/repo/PopularMoviesDataRepositoryTest.kt

```diff
@@ -15,65 +16,55 @@ import kotlinx.coroutines.test.runTest
 import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DynamicTest
 import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestFactory
 
 internal class PopularMoviesDataRepositoryTest {
 
-    val moviesEntity = listOf(
-        MovieEntity(10001, "Test1", 5.4, "2022-06-23", "https://image.tmdb.org/t/p/w500/something1.com", false),
-    )
-    val moviesModel = listOf(
-        MovieModel(10001, "Test1", 5.4, "2022-06-23", "https://image.tmdb.org/t/p/w500/something1.com", false),
-    )
-    val movieResponseRemote: MovieResponseRemote = mockk()
     val apiService: MoviesApi = mockk()
     val dao: MovieDao = mockk(relaxed = true)
     val tested = PopularMoviesDataRepository(apiService, dao)
-    val resultThrowable = Throwable(
-        "Unable to resolve host \"api.themoviedb.org\": No address associated with hostname"
-    )
 
     @BeforeEach
     fun beforeEach() {
         mockkStatic(List<MovieEntity>::toDomain)
+        mockkStatic(MovieModel::toEntity)
     }
 
     @AfterEach
     fun afterEach() {
         unmockkAll()
     }
 
-    @Test
-    fun `SHOULD map to the MovieEntity type WHEN method get data from api`() =
-        runTest {
-            coEvery { apiService.getPopularMovies(any()) } returns movieResponseRemote
-            coEvery { dao.getMovies() } returns moviesEntity
+    @TestFactory
+    fun `SHOULD return specific value WHEN in database is specific value`() =
+        listOf(
+            emptyList(),
+            listOf(mockk<MovieModel>())
+        ).map { movies ->
+            DynamicTest.dynamicTest("SHOULD return + $movies WHEN in database is $movies") {
```

Comment:
the name of the test is quite not understandable for me...

Reply:
Done 😄